File: packages/csharp/HtmlToMarkdown/Metadata/ProcessingWarning.cs
using System.Text.Json.Serialization;

namespace HtmlToMarkdown.Metadata;

/// <summary>
/// A non-fatal processing warning produced during conversion.
/// </summary>
public record ProcessingWarning
{
    /// <summary>
    /// Human-readable description of the warning.
    /// </summary>
    [JsonPropertyName("message")]
    public string Message { get; init; } = string.Empty;

    /// <summary>
    /// The warning category (e.g., "malformed_html", "encoding_fallback").
    /// </summary>
    [JsonPropertyName("kind")]
    public string Kind { get; init; } = string.Empty;
}

File: packages/java/src/main/java/dev/kreuzberg/htmltomarkdown/ProcessingWarning.java
package dev.kreuzberg.htmltomarkdown;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * A non-fatal processing warning produced during conversion.
 *
 * @param message human-readable description of the warning
 * @param kind the warning category (e.g., "malformed_html", "encoding_fallback")
 */
public record ProcessingWarning(
    @JsonProperty("message") String message, @JsonProperty("kind") String kind) { }
